Update Promise.try description (#35569)
mdn/content: files/en-us/web/javascript/reference/global_objects/promise/try/index.md

## Description

> You may have an API that takes a callback. The callback may be synchronous or asynchronous. You want to handle everything uniformly by wrapping the result in a promise. The most straightforward way might be {{jsxref("Promise/resolve", "Promise.resolve(func())")}}. The problem is that if `func()` synchronously throws an error, this error would not be caught and turned into a rejected promise. The correct way to do so is the following, which `Promise.try(func)` is equivalent to:

Reviewer: I'm fine to remove this, but FYI:

`Promise.try(func)` is exactly equivalent to `new Promise((resolve) => resolve(func()));`, but this doesn't mean that the `try()` method is implemented using `Promise.try = () => new Promise((resolve) => resolve(func()))`. It's implemented using the second code snippet. It just happens that when you call `try` on `Promise` instead of a subclass, the `Promise` constructor has inbuilt error catching.

Reviewer: You are not wrong, but "behavior" is the only thing observable about built-in methods, so if two code snippets have the same behavior then they are exactly equivalent.

Author: It is still confusing to call something "exactly equivalent" if the text representations that the audience reads are different.

You may have an API that takes a callback. The callback may be synchronous or asynchronous. You want to handle everything uniformly by wrapping the result in a promise. The most straightforward way might be {{jsxref("Promise/resolve", "Promise.resolve(func())")}}. The problem is that if `func()` synchronously throws an error, this error would not be caught and turned into a rejected promise. The correct way to do so is the following, which `Promise.try(func)` is equivalent to:

```js
new Promise((resolve) => resolve(func()));
```

> Except that `Promise.try()` is perhaps more concise and readable. To be more exact, the following is a more faithful representation of the implementation of `Promise.try()` (although it should still not be used as a polyfill):

Reviewer: Please keep the "although it should still not be used as a polyfill" part. It is our responsibility to never provide polyfills because we cannot guarantee 100% spec compatibility. In this case, the pitfall is that someone may overwrite `globalThis.Promise`, so the following is not a faithful polyfill:

```js
Promise.try = () => new Promise((resolve, reject) => {
  try {
    resolve(func());
  } catch (error) {
    reject(error);
  }
});
```

Instead, you must do:

```js
const Promise = globalThis.Promise;

Promise.try = () => ...
```

And then there's the problem of making `try` configurable, non-enumerable, writable, and make the `toString` result of `try` correct, etc., etc. We'd just like to not involve ourselves with polyfilling business and keep the example strictly for demonstration.

> For the built-in `Promise()` constructor, errors thrown from the executor are automatically caught and turned into rejections, so the above would also be equivalent to:

Reviewer: ```suggestion
For the built-in `Promise()` constructor, errors thrown from the executor are automatically caught and turned into rejections. An equivalent implementation that generalizes to any `Promise` subclass is:
```
